master/cache.lua

```diff
@@ -43,7 +71,7 @@ local function request(host, port, path)
         return nil, err
     end
 
-    if res.status ~= 200 then
+    if res.status ~= 200 and accept_404_reply and res.status ~= 404 then
```

Comment:
Actually there are two bugs here - one that you found and the other in the LineBufferFilter class that prevented test from detecting this one :(

Thanks a lot for catching it!